### PR TITLE
remove tooltip illegal character

### PR DESCRIPTION
**Pull request by simonleung8**
Currently throughput tooltip uses command name as class name, and it might contain illegal characters (e.g. gcf:push).

This change make sure illegal character will not end up in class name
no test provided because I don't think it is necessary to test logic within private methods

### CODE REVIEW

**Comment by julz** [comment]
It's necessary to test the public effect. For example that tooltips are generated with a valid classname even when an experiment has an illegal character.

**Comment by simonleung8** [comment]
You are right.  Backfilled the fix with test

File: ui/js/tests.js [hunk 1]
```diff
@@ -254,6 +254,19 @@ describe("Throughput chart", function() {
     expect( $(node).find("g.datalogin circle")[0].getAttribute("fill") ).toEqual(color)
   })
 
+  it("it should remove illegal characters in tooltips class name", function() {
```
Comment (julz) [inline ui/js/tests.js]: are there other illegal chars we should remove while we're thinking about this?

File: ui/js/tests.js [hunk 2]
```diff
@@ -254,6 +254,19 @@ describe("Throughput chart", function() {
...
+    var illegalName = "gcf:login";
+    var workload = [{ Commands: { illegalName: {"Count": 1, "Throughput": 0.50}}}];
+
+    chart(workload);
+    d3.select(node).select("path.line").on("mouseover")({cmd: illegalName, throughput:[0.50]});
+
+    expect( $(node).find("g.data" + illegalName).length ).toEqual(0);
+    
+    var legalName = illegalName.replace(":","");
```
Comment (julz) [inline ui/js/tests.js]: why repeat the logic in the test here? you know that the legal version of the name is "g.datagcflogin"

Comment (simonleung8) [inline ui/js/tests.js]: I was thinking it is easier to read, anyway, took out the repeating logic

File: ui/js/tests.js (hunk 1 above)
Comment (simonleung8) [inline ui/js/tests.js]: Now it checks for all non A..Z 0..9 chars, and replace them with "_"

File: ui/js/throughput.js [hunk 3]
```diff
@@ -167,12 +167,12 @@ d3_throughput = function() {
   function drawToolTip(d, c) {
     const pointRadis = 13;
     const yOffset = 4;
-    var tooptip = svg.selectAll("g.data" + d.cmd).data(d.throughput).enter()
+    var tooptip = svg.selectAll("g.data" + d.cmd.replace(/[^a-zA-Z0-9]/g,"_")).data(d.throughput).enter()
```
Comment (julz) [inline ui/js/throughput.js]: lets remove the duplication of the regex? otherwise, looks ok

**Comment by simonleung8** [comment]
No more duplication of regex.

**Pull request merged**